uxarray/grid.py

```diff
@@ -528,6 +531,33 @@ def _build_edge_node_connectivity(self):
         setattr(self, "Mesh2_edge_nodes", self.ds['Mesh2_edge_nodes'])
         setattr(self, "nMesh2_edge", edge_nodes_unique.shape[0])
 
+    def build_face_edges_connectivity(self):
+        """Build the `Grid.face_edges_connectivity`, a DataArray of indices
+        indicating edges that are neighboring each face.
+
+        The `Grid.face_edges_connectivity` will be an integer DataArray
+        of size (nMesh2_face, MaxNumNodesPerFace) (optional)
+        """
+        if "Mesh2_edge_nodes" not in self.ds.keys() or (
+                self.source_grid != "From vertices" and
+                self.mesh_type == "mpas"):
+            self._build_edge_node_connectivity(repopulate=True)
```

Comment:
> Before entering `self._build_edge_node_connectivity()`, we need to determine whether we need to repopulate the edge nodes, which is when `Mesh2_edge_nodes` is not in our dataset. The only time we need to repopulate is when we already have the edge nodes, but they don't contain the inverse indices. We could do something like this inside of the grid constructor:
> 
> ```python
> # determine if we need to repopulate our edge nodes
> if "Mesh2_edge_nodes" not in self.ds:
>     repopulate = True    
> else:
>     repopulate = False
>       
> # construct edge node connectivity 
> self._build_edge_node_connectivity(repopulate) 
>  
> # construct face edge connectivity
> self._build_face_edge_connectivity()
>   
> # build face dimension, possibly safeguard for large datasets 
> self._build_face_dimension() 
> ```

1. The `repopulate` flag, as indicated in its name, it's for the case where the `Grid.edge_nodes_connectivity` exists but the `inverse_indices` doesn't. So this change will be unnecessary. Imagine when we read-in the MPAS data:
It has the  "Mesh2_edge_nodes", so `repopulate = False`, making this entire block equivalent to the initial implementation
```python
if "Mesh2_edge_nodes" not in self.ds:
    self._build_edge_node_connectivity()
```

2. I put the check in the `Grid.build_face_nodes_connectivity` by checking if the source is MPAS. So that if the user doesn't need the `Grid.Build_face_edges_connectivity`, it won't be repopulated again

3. `Grid.Build_face_edges_connectivity` is just a optional API, indicating that not everyone needs the "Mesh2_face_edges", so we don't wanna waste time calling it in the initialization. We will only call it when needed